bc_obps/common/permissions.py

```diff
@@ -1,51 +1,118 @@
+                    UserOperator.objects.exclude(status=UserOperator.Statuses.DECLINED)
+                    .only('role')
+                    .get(user=user.user_guid)
+                )
+                user_operator_role = user_operator.role
+            except UserOperator.DoesNotExist:
+                pass
+            if not user_operator_role or user_operator_role not in authorized_user_operator_roles:
+                raise HttpError(401, UNAUTHORIZED_MESSAGE)
+
 
 """
 NOTE: `industry_user_must_be_approved` is an optional parameter that defaults to True in `raise_401_if_user_not_authorized`
 and this is the reason why it is not included in some of the permission configurations.
 """
 
-all_authorized_app_roles = AppRole.get_all_authorized_app_roles()
-all_industry_user_operator_roles = UserOperator.get_all_industry_user_operator_roles()
-permission_configs: Dict[str, Dict] = {
-    "all_roles": {
-        'authorized_app_roles': AppRole.get_all_app_roles(),
-        'authorized_user_operator_roles': all_industry_user_operator_roles,
-        'industry_user_must_be_approved': False,
-    },
-    "authorized_roles": {
-        'authorized_app_roles': all_authorized_app_roles,
-        'authorized_user_operator_roles': all_industry_user_operator_roles,
-        'industry_user_must_be_approved': False,
-    },
-    "approved_authorized_roles": {
-        'authorized_app_roles': all_authorized_app_roles,
-        'authorized_user_operator_roles': all_industry_user_operator_roles,
-    },
-    "authorized_irc_user_and_industry_admin_user": {
-        'authorized_app_roles': all_authorized_app_roles,
-        'authorized_user_operator_roles': ["admin"],
-    },
-    "industry_user": {
-        'authorized_app_roles': ["industry_user"],
-        'authorized_user_operator_roles': all_industry_user_operator_roles,
-        'industry_user_must_be_approved': False,
-    },
-    "approved_industry_user": {
-        'authorized_app_roles': ["industry_user"],
-        'authorized_user_operator_roles': all_industry_user_operator_roles,
-    },
-    "approved_industry_admin_user": {
-        'authorized_app_roles': ["industry_user"],
-        'authorized_user_operator_roles': ["admin"],
-    },
-    "authorized_irc_user": {
-        'authorized_app_roles': AppRole.get_authorized_irc_roles(),
-    },
-}
+
+def get_permission_configs(permission: str) -> Optional[Union[Dict[str, List[str]], Dict[str, bool]]]:
+    """
+    Returns the permission configurations for the specified permission.
+    :param permission: The permission to get the configuration for.
+    :return: The permission configurations.
+
+    NOTE: We must make this a function so we don't hit database when collecting tests.(we don't have access to DB during collection)
+    """
+    all_authorized_app_roles = AppRole.get_all_authorized_app_roles()
+    all_industry_user_operator_roles = UserOperator.get_all_industry_user_operator_roles()
+    permission_configs: Dict[str, Dict] = {
```

Comment:
I think these queries will now run every single time we authenticate a request, so all the time. Is this intentional?

Reply:
Yes, I added the reason as a note in the docstring of this function. However, I believe we can cache it to prevent calling it every time.

Reply:
Would it be sensible to cache this for one hour? I don't believe we update the AppRole and UserOperator roles very often.